EIPS/eip-5027.md

```diff
@@ -0,0 +1,91 @@
+---
+eip: 5027
+title: Remove the limit on contract code size
+description: Change the limit on contract size from 24576 to infinity
+author: Qi Zhou (@qizhou)
+discussions-to: https://ethereum-magicians.org/t/eip-5027-unlimit-contract-code-size/9010
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-04-21
+requires: 170
+---
+
+
+## Abstract
+
+Remove the limit on the contract code size, i.e., only limit the contract code size by block gas limit, with minimal changes to existing code and proper gas metering adjustment to avoid possible attacks.
+
+
+## Motivation
+
+The motivation is to remove the limit on the code size so that users can deploy a large-code contract without worrying about splitting the contract into several sub-contracts.
+
+With the dramatic growth of dApplications, the functionalities of smart contracts are becoming more and more complicated, and thus, the sizes of newly developed contracts are steadily increasing.  As a result, we are facing more and more issues with the 24576-bytes contract size limit.  Although several techniques such as splitting a large contract into several sub-contracts can alleviate the issue, these techniques inevitably increase the burden of developing/deploying/maintaining smart contracts.
+
+The proposal implements a solution to remove the existing 24576-bytes limit of the code size.  Further, the proposal aims to minimize the changes in the client implementation (e.g., Geth) with
+- proper gas metering to avoid abusing the node resources for contract-related opcodes, i.e, `CODESIZE/CODECOPY/EXTCODESIZE/EXTCODECOPY/EXTCODEHASH/DELEGATECALL/CALL/CALLCODE/STATICCALL/CREATE/CREATE2`; and
```

Comment:
The first reference to any OPCODE must include its number in the form `OPCODE_NAME (0x00)`.  See https://eips.ethereum.org/EIPS/eip-1#special-requirements-for-core-eips for details.

Reply:
Done.  Glad to know.  Thanks for the comment.